extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java

```diff
@@ -308,13 +309,31 @@ void additionalBean(BuildProducer<AdditionalBeanBuildItem> additionalBeanProduce
 
     @BuildStep
     OpenApiFilteredIndexViewBuildItem smallryeOpenApiIndex(CombinedIndexBuildItem combinedIndexBuildItem,
-            BeanArchiveIndexBuildItem beanArchiveIndexBuildItem) {
-        CompositeIndex compositeIndex = CompositeIndex.create(combinedIndexBuildItem.getIndex(),
+            BeanArchiveIndexBuildItem beanArchiveIndexBuildItem,
+            BuildExclusionsBuildItem buildExclusionsBuildItem) {
```

Comment:
I couldn't find `BuildExclusionsBuildItem` used anywhere else in Quarkus, but it seems like the perfect fit for this fix...